src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp

```diff
@@ -233,65 +244,51 @@ class TDatabaseFeedManager final : public Observer<nlohmann::json&>
                         throw std::runtime_error("Invalid line. file: " + path.get_ref<const std::string&>());
                     }
 
-                    parsedLine["resource"] = parsedLine["name"];
+                    parsedLine["resource"] = parsedLine.at("name");
                     parsedLine["type"] = "create";
 
-                    orchestration(parsedLine, m_feedDatabase.get());
+                    resourceProcessing(parsedLine, feedDatabase);
                 }
             }
 
-            m_feedDatabase->flush();
-
-            // Update the offset.
-            contentManagerUpdateOffset(topicName, parsedMessage.at("offset"));
-
-            if (parsedMessage.contains("/fileMetadata/hash"_json_pointer))
-            {
-                // Update last parsed file hash.
-                contentManagerUpdateHash(topicName,
-                                         parsedMessage.at("fileMetadata").at("hash").get_ref<const std::string&>());
-            }
+            const auto offset = parsedMessage.at("offset").get<int>();
+            const auto hash = std::string(parsedMessage.at("fileMetadata").at("hash").get<const std::string>());
```

Comment:
When I try to process a snapshot it shows me the following error, which I don't think is correct because the content “fileMetadata” is only published when a file is downloaded. It may be associated with this line.
![2024-09-13_09-10](https://github.com/user-attachments/assets/89e8ae2c-440e-401d-9c81-96ab479512b2)

Reply:
Good catch, I've restored the old logic of verifying if the fileMetadata field exists